### ISSUE TITLE
How to do bottle deposits?

### ISSUE THREAD

**Issue by scannerdarkly** [CORE-POS/IS4C#634]
I understand that bottle deposits are entered against a special store department designated in the installation setup. I imagine the same department is used for both incoming and outgoing bottles.

Are the return deposits supposed to be entered as Refunds against that designated department? If so, is there a way to have this particular type of refund be exempt from the normal manager password protection scheme?

Or should we set up a specific PLU that has a negative purchase price? Or some other scheme?

Related, is there a way to have a water purchase include the deposit amount by default, then let the cashier specify a returned bottle count to bring the purchase price back down? I could imagine this being the most efficient workflow in certain situations.

**Comment by gohanman** [comment]
I think there are at least a couple different ways to do it.

Option 1:
Create one or more deposit PLUs with negative prices. Enter the PLUs in the "Deposit" field for items that require a deposit. When ringing in the item the deposit PLU gets added automatically. They should stay linked on voids, too.

Option 2:
Create a single deposit PLU with price $0. Put it in the department that's configured on the lane as the bottle deposit department. After entering the PLU the cashier will be prompted for a price. This will actually occur with any $0 item; all the deposit department setting does is negate the amount so the cashier doesn't need to type "-".

There's nothing to prompt for a returned bottles. I'm sure it could be done but we just have a bottle return key.

**Comment by scannerdarkly** [comment]
So in the first case, the PLU with the negative price should be in the Deposit department as well? Or is that only for the second case?

And can the cashier in general type a negative price for any open ring situation? Or is this a special case for just the deposit department? (Just thinking about security implications here.)

**Comment by gohanman** [comment]
The deposit department has no effect in the first case. The second case is
not an open ring; it's a price override. Price overrides on items in the
deposit department get multiplied by -1.

Separately, yes, negative open rings are allowed. Whether the cashier can
type one does of course depend whether there's a "-" key on the keyboard.

**Comment by scannerdarkly** [comment]
I attempted the first approach. Positive price of $7.49 on 5-gallon spring water (PLU 351), linked to PLU 350 bottle deposit which has -$7.00 price. Both in the basic grocery department, though the deposit PLU is marked non-discountable. When I ring it in, it shows net $0.49. Do you think there’s something I’m missing here?

**Comment by joelbrock** [comment]
The math seems right.  $7.49 - $7.00 == $0.49.  
You could put the deposit into a dedicated bottle deposits department.  Or you could assign it the same dept as the item, which have done here.  It all depends on how you want the deposits to appear on the reporting-side.

**Comment by gohanman** [comment]
I think I misspoke. All my bottle deposit PLUs are positive. There's really only the one way of doing automatic deposits.

The bottle return department is the other half of the transaction. You can use a zero-priced PLU in the bottle return department so price overrides are negated, or you could just have negative priced return PLUs. Or I guess you could do refunds with the original deposit PLUs.

**Comment by gohanman** [comment]
This may be a slightly more accurate recap:
https://github.com/CORE-POS/IS4C/wiki/Automatic-Bottle-Deposits

**Comment by gohanman** [comment]
Debugging returns with another store revealed there are two return related settings that can interfere with one another. I attempted to resolve them 0b1987bcb21cca5c32b0c3d8580c7cb8ea45726d so that the **BottleReturnDept** setting on **Scanning** will handle both open rings to return departments and zero-priced items in return departments. I'll leave the **Extras** tab setting in place for backwards compatibility. I've updated the doc page to cover how things should be configured in current versions.
https://github.com/CORE-POS/IS4C/wiki/Automatic-Bottle-Deposits
